code/game/machinery/shuttle/custom_shuttle.dm

```diff
@@ -0,0 +1,60 @@
+/obj/machinery/shuttle
+	name = "shuttle component"
+	desc = "Something for shuttles."
+	density = TRUE
+	obj_integrity = 250
+	max_integrity = 250
+	icon = 'icons/turf/shuttle.dmi'
+	icon_state = "burst_plasma"
+	idle_power_usage = 150
+	circuit = /obj/item/circuitboard/machine/shuttle/engine
+	var/icon_state_closed = "burst_plasma"
+	var/icon_state_open = "burst_plasma_open"
+	var/icon_state_off = "burst_plasma_off"
+
+/obj/machinery/shuttle/Initialize()
+	. = ..()
+	GLOB.custom_shuttle_machines += src
+	check_setup()
+
+//Call this when:
+// - The shuttle it's attached to gets 'Calculate Stats' called
+// - A heater next to this object gets wrenched into place
+// - A heat next to this gets wrenched out of place
+// - This gets wrenched into place
+/obj/machinery/shuttle/proc/check_setup(var/affectSurrounding = TRUE)
+	if(!affectSurrounding)
+		return
+	//Don't update if not on shuttle, to prevent lagging out the server in space
+	if(!istype(get_turf(src), /area/shuttle/custom))
+		return
+	//Check the standard machines
+	for(var/obj/machinery/shuttle/shuttle_machine in GLOB.custom_shuttle_machines)
+		if(!shuttle_machine)
+			continue
+		if(shuttle_machine == src)
+			continue
+		shuttle_machine.check_setup(FALSE)
```

Comment:
So every shuttle machine calls every other shuttle machine's check_setup()? Seems like only a single object should be responsible for calling every machine's check_setup().

Also, shouldn't it only run on machines that are on the same shuttle?

Reply:
The only thing this is used for is checking if an engine has a heater.
Rather then checking everything, I am going to only check the turf that the engine should be on relative to the caller